planner/core/src/main/java/eu/seaclouds/platform/planner/core/Planner.java

```diff
@@ -167,17 +174,19 @@ public String rePlan() throws ParsingException, IOException {
         List<NameValuePair> optParams = getReqParams(
                 new BasicNameValuePair("aam", aam),
                 new BasicNameValuePair("dam", dam),
-                new BasicNameValuePair("mmResults", mmOutput)
+                new BasicNameValuePair("offers", mmOutput)
         );
-        String optResult = optimizerClient.postRequest("/optimize", optParams);
-        return optResult; //TODO: check the output for the optimizer
+
+        Pair<String, String> optResult = optimizerClient.postRequest("/optimize", optParams);
+        if(optResult.first.equals("200")){
+            return optResult.second; //TODO: check the output for the optimizer
+        }
+        return null; //TODO: return failure
     }
 
     public String generateDam() {
 
-        List<NameValuePair> damInfoParams = getReqParams(new BasicNameValuePair("adp", adp));
-
-        String monitorInfo = new HttpHelper(monitorGenURL).postRequest(DAM_GEN_OP, damInfoParams);
+        String monitorInfo = new HttpHelper(monitorGenURL).postInBody(MONITOR_GEN_OP, adp);
```

Comment:
monitorInfo could be a parameter, but since it is used just there there is no need to promote it. Any ideas?
The generator returns a JSon String, since. Non need to add a specific POJO for that.